File: typescript/src/arrays/two-sum/two-sum.ts
/**
 * Problem: Two Sum
 * Difficulty: Easy
 * Time Complexity: O(N)
 * Space Complexity: O(N)
 */
export const twoSum = (nums: number[], target: number) => {
  const hashmap: Record<number, number> = {};

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];
    if (diff in hashmap) {
      return [hashmap[diff], i];
    }
    hashmap[nums[i]] = i;
  }
  return [-1, -1];
};

File: src/arrays/two_sum/two_sum.py
"""
Problem: Two Sum
Difficulty: Easy
Time Complexity: O(N)
Space Complexity: O(N)
"""
from typing import List

def two_sum(nums: List[int], target: int) -> List[int]:
    hashmap = {}

    for i in range(len(nums)):
        diff = target - nums[i]
        if diff in hashmap:
            return [hashmap[diff], i]
        hashmap[nums[i]] = i
    return [-1, -1]
